vpr/src/place/place.cpp

```diff
@@ -758,6 +935,42 @@ void try_place(const t_placer_opts& placer_opts,
     }
     auto post_quench_timing_stats = timing_ctx.stats;
 
+    //Final timing analysis
+    PlaceCritParams crit_params;
+    crit_params.crit_exponent = state.crit_exponent;
+    crit_params.crit_limit = placer_opts.place_crit_limit;
+
+    if (placer_opts.place_algorithm.is_timing_driven()) {
+        perform_full_timing_update(crit_params,
+                                   place_delay_model.get(),
+                                   placer_criticalities.get(),
+                                   placer_setup_slacks.get(),
+                                   pin_timing_invalidator.get(),
+                                   timing_info.get(),
+                                   &costs);
+        VTR_LOG("post-quench CPD = %g (ns) \n", 1e9 * timing_info->least_slack_critical_path().delay());
+    }
+
+    //See if our latest checkpoint is better than the current placement solution
+    if (placer_opts.place_checkpointing && placement_checkpoint.cp_is_valid() && timing_info->least_slack_critical_path().delay() > placement_checkpoint.get_cp_cpd() && costs.bb_cost < 1.05 * placement_checkpoint.get_cp_bb_cost()) {
```

Comment:
Good spot for a helper function:  restore_best_placement () or some such.

Reply:
Done